server/src/models/user.js

```diff
@@ -653,6 +653,42 @@ module.exports = {
 		});
 	},
 
+	checkMapNotify: (followeeID, mapID) => {
+		return MapNotify.findOne({
+			where: { followeeID: followeeID, mapID: mapID },
+			raw: true
+		})
+	},
+
+	updateMapNotify: (followeeID, mapID, notify) => {
+		return MapNotify.findOne({
+			where: {
+				followeeID: followeeID,
+				mapID: mapID
+			}
+		}).then(map => {
+			if (map) {
+				return MapNotify.update(
+					{ notifyOn: notify },
+					{ where: { followeeID: followeeID, mapID: mapID } }
+				)
+			} else {
+				return MapNotify.create({
+					notifyOn: notify,
+					mapID: mapID,
+					followeeID: followeeID
+				},
+				)
+			}
+		})
```

Comment:
If I recall correctly, models have a `findOrCreate` extension method on them that returns an object with the model and a boolean of if it created or not. An example can be found with the run submission, specifically PB runs:

https://github.com/momentum-mod/website/blob/staging/server/src/models/run.js#L293-L318

Reply:
And if it returns the model, you should use the model directly and not fetch it again from the DB:
```js
}).spread( (mapNotify, created) => {
    if (!created) {
        return mapNotify.update({ notifyOn: notify });
    }
    return Promise.resolve();
}
```